File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/repositories/VehicleModelRepository.java
package com.example.demo.repositories;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.example.demo.dto.VehicleModelDTO;
import com.example.demo.entities.VehicleModel;

public interface VehicleModelRepository extends JpaRepository<VehicleModel, Integer> {

	@Query("SELECT new com.example.demo.dto.VehicleModelDTO(" +
		       "v.modelId, v.modelName, v.minimumQuantity) " +
		       "FROM VehicleModel v " +
		       "WHERE v.manufacturer.manufacturerId = :manufacturerId " +
		       "AND v.segment.segmentId = :segmentId")
		List<VehicleModelDTO> getModelsByManufacturerAndSegment(
		        @Param("manufacturerId") int manufacturerId,
		        @Param("segmentId") int segmentId);
}

File: backend-dotnet/Repository/VehicleModelRepository.cs

using backend_dotnet.Data;
using backend_dotnet.DTO;
using backend_dotnet.Models;
using Microsoft.EntityFrameworkCore;

namespace backend_dotnet.Repositories
{
    public class VehicleModelRepository : IVehicleModelRepository
    {
        private readonly ApplicationDbContext _context;

        public VehicleModelRepository(ApplicationDbContext context)
        {
            _context = context;
        }

        public VehicleModel? GetById(int modelId)
        {
            return _context.VehicleModels
                .Include(v => v.Manufacturer)
                .Include(v => v.Segment)
                .FirstOrDefault(v => v.ModelId == modelId);
        }

        public List<VehicleModelDTO> GetModelsByManufacturerAndSegment(
            int manufacturerId,
            int segmentId)
        {
           
return _context.VehicleModels
    .Where(v =>
        v.ManufacturerId == manufacturerId &&
        v.SegmentId == segmentId)
    .Select(v => new VehicleModelDTO
    {
        ModelId = v.ModelId,
        ModelName = v.ModelName,
        MinimumQuantity = v.MinimumQuantity
    })
    .ToList();

        }
    }
}

